### PR TITLE
Cohort Diagnostics organization and export underlying functions

### PR DESCRIPTION
**Pull request by mdlavallee92**
## Purpose
Goal of pull request is to provide documentation to key underlying functions of cohort diagnostics that sit underneath the global execute ``executeDiagnostics``. Also helped organized functions to provide more UI consistency. Create Unit tests for newly exported functions (TODO). 

### 1) Unify terminology for cohort diagnostic functions:

get (single cohort) => batch (multiple cohorts) => execute (global run)
Function signatures were changed to match these key verbs.

### 2) make function arguments consistent
some examples:
cohorts input changed to cohortDefinitionSet (since this is the input in the execute)
covariateSettings changed to temporalCovariateSettings

Also organized signatures:
batch_fn args example:
connection
schemas(cdm, vocab, temp, cohort)
cohortTable
cohortDefinitionSet
cdmVersion (if needed)
Export (databaseId, exportFoldr, minCellCount) ->match ``makeDataExportable``
Batch (instantiatedCohorts, recordKeepingFile, incremental)  -> match ``recordTasksDone`` for incremental mode
remaining function options

## 3) Unit Tests 
In progress 🏗️ 

## Work Zone 🚧 
Files where work was done:
1) Cohort Counts (CohortLevelDiagnostics.R)
2) Inclusion Rules (InclusionRules.R)
3) Concept Sets (ConceptSets.R)
4) Time Series (TimeSeries.R)
5) Visit Context (VisitContext.R)
6) Incidence Rates (IncidenceRates.R)
7) Cohort Overlap (CohortRelationship.R)
8) Temporal Characterization (CohortCharacterizationDiagnostics.R)  
9) executeDiagnostics (RunDiagnostics.R) -> the global execute function

### CODE REVIEW

**Review by azimov — COMMENTED** [review]
My only query is if we want to export `batch` functions - my initial thought was that we would expose getter style functions as the internals of how the processing takes place isn't especially important.

No other issues so far - this will be super useful.

**Comment by mdlavallee92** [comment]
@azimov for the unit tests can we create the cohort definitions once in the setup file and use them for all of the tests? Is there a particular reason the cohorts are generated across multiple tests? 

Also I changed the functions so that only those that actually run in batch are called `batch` (removed the export). and functions that are just iterative wrappers are called `generate`. Let me know if that is ok

**Comment by azimov** [comment]
> @azimov for the unit tests can we create the cohort definitions once in the setup file and use them for all of the tests? Is there a particular reason the cohorts are generated across multiple tests?

I'd need to check each individual test to be certain but this is almost certainly a handover from when this package also generated the cohorts in place of `CohortGenerator`. If you want to change this feel free, it would be a big improvement and I'd certainly try to structure your tests this way if you can.

File: R/CohortCharacterizationDiagnostics.R [hunk 1]
```diff
@@ -22,7 +47,7 @@ getCohortCharacteristics <- function(connectionDetails = NULL,
                                      cohortTable = "cohort",
                                      cohortIds,
                                      cdmVersion = 5,
-                                     covariateSettings,
+                                     temporalCovariateSettings,
```
[the remark this answers is not in the document]
Comment (azimov) [reply R/CohortCharacterizationDiagnostics.R:50]: Agreed, I think a global batch call would be a good simplification but these functions each have a tonne of unique side effects

**Comment by mdlavallee92** [comment]
@azimov closing since this is too out of date. will start fresh

**Pull request closed without merging**